stellarphot/photometry/source_detection.py

```diff
@@ -141,52 +145,130 @@ def compute_fwhm(ccd, sources, fwhm_estimate=5,
 
-    find_fwhm : bool, optional
+    find_fwhm : bool, optional (default=True)
         If ``True``, estimate the FWHM of each source by fitting a 2D Gaussian
-        to it. If ``False``, estimate the FWHM of each source by computing
-        the second moments of its light distribution.
+        to it. If ``False``, the 'fwhm_x', 'fwhm_y', and 'width' columns of
+        output table will be filled with NaN values.
 
-    sky_per_pix_avg : float or array-like of float
-        Sky background to subtract before centroiding.
+    sky_per_pix_avg : float or None, optional (default=None)
+        Sky background to subtract before centroiding. If set to ``None``,
+        it will be estimated using the mean of the sigma_clipped_stats
+        of the image.
 
     Returns
     -------
 
-    sources: `astropy.table.Table`
+    sources: `stellardev.SourceListData`
         A table of the positions of sources in the image.  If `find_fwhm` is
-        ``True``, includes a column called ``FWHM``.
+        ``True``, includes columns for `fwhm_x`, `fwhm_y, and `width`.  If
+        the input CCDData object has WCS information, the columns `ra` and
+        `dec` are also populated, otherwise they are set to NaN.
     """
+
+    # If user uses units for fwhm or sky_per_pix_avg, convert to value
+    if isinstance(sky_per_pix_avg, u.Quantity):
+        sky_per_pix_avg = sky_per_pix_avg.value
+    if isinstance(fwhm, u.Quantity):
+        fwhm = fwhm.value
+
+    # Get statistics of the input image (and use them to estimate the sky background
+    # if not provided).  Using clipped stats should hopefully get rid of any
+    # bright stars that might be in the image, so the mean should be a good
+    # estimate of the sky background.
     mean, median, std = sigma_clipped_stats(ccd, sigma=sigma, maxiters=iters)
-    daofind = DAOStarFinder(fwhm=fwhm, threshold=threshold * std)
-    sources = daofind(ccd - median)
-    print(sources)
+    print(f"source_detection: mean={mean:.4f}, median={median:.4f}, std={std:.4f}")
+    if sky_per_pix_avg is None:
+        sky_per_pix_avg = mean
+        print(f"source_detection: sky_per_pix_avg set to {sky_per_pix_avg:.4f}")
+
+    # Identify sources applying DAOStarFinder to a "sky subtracted"
+    # image.
+    print(f"source_detection: threshold set to {threshold}* standard deviation "
+          f"({std:.4f})")
+    print(f"source_detection: Assuming fwhm of {fwhm} for DAOStarFinder")
+    daofind = DAOStarFinder(fwhm = fwhm, threshold = threshold * std)
+    sources = daofind(ccd - sky_per_pix_avg)
+    # daofind should be run on background subtracted image
+    # (fails, or at least returns garbage, if sky_per_pix_avg is too low)
+    src_cnt = len(sources)
+    print(f"source_detection: {src_cnt} sources identified.")
+    #print(sources)
+
+    # If image as WCS, compute RA and Dec of each source
+    try:
+        sources['ra'], sources['dec'] = ccd.wcs.all_pix2world(sources['xcentroid'],
```

Comment:
The preferred way to call this has changed. It is now

```suggestion
        sources['ra'], sources['dec'] = ccd.wcs.pixel_to_world(sources['xcentroid'],
```

Reply:
This may also change the return type too -- it may be coordinate objects rather than separate RA and Dec

Reply:
OK, I fixed this.